packages/components/src/components/SquiggleEditor.tsx

```diff
@@ -50,7 +61,8 @@ export let SquiggleEditor: React.FC<SquiggleEditorProps> = ({
   diagramCount,
   onChange,
   environment,
-  bindings = {},
+  bindings = defaultBindings,
+  parameters = defaultParameters,
```

Comment:
It's fairly awkward how we now need to pass in squiggleString+bindings+paramaters+samplingInputs (needed soon), separately, in each case.

A different way of doing it would be to have one typed ``squiggleInputs`` object, which requires each of these. This larger object could have its own defaults or whatever.

Reply:
This is a good call. That being said, that kind of already is the interface, as props is already an object. The public facing interface looks more like `renderSquiggleEditor({initialSquiggleString: "normal(0, 1)"})` plus adding whatever props you want.

That being said, I absolutely feel you when talking about `run` and `runPartial` in src/js/index.ts. Getting to 4 different parameters when often you might only care about 1 or 2 does get to be a bit annoying. I think you might be able to start to see some of the frustration in `__tests__/TS/JS_test.ts`

Reply:
I'd go so far as to use the `newtype` pattern around `Object` for `parameters` (or `variables`), and have run be of signature 
```rescript
run : (string, opts) => result<squiggleResult, squiggleError>
```
where 
```rescript
type bindings = ...
type variables = Object
type ops = {bindings: bindings, env: env, variables: variables} 
```

Maybe `run : (opts, string) => result<...>` instead, idk.